app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java

```diff
@@ -350,6 +352,10 @@ protected void runRegexReplaceAction(final List<ReplacePattern> patterns, final
 
             lineStart = StringUtils.getLineEnd(text, lineStart, selEnd) + 1;
         }
+
+        _hlEditor.setSelection(
+                StringUtils.getIndexFromLineOffset(text, lStart),
+                StringUtils.getIndexFromLineOffset(text, lEnd));
```

Comment:
This logic + the code in `StringUtils` preserves selection when action is pressed